templates/solochain/runtime/src/lib.rs

```diff
@@ -195,57 +189,40 @@ impl pallet_grandpa::Config for Runtime {
 	type MaxNominators = ConstU32<0>;
 	type MaxSetIdSessionEntries = ConstU64<0>;
 
-	type KeyOwnerProof = sp_core::Void;
+	type KeyOwnerProof = Void;
 	type EquivocationReportSystem = ();
 }
 
+// Implements the types required for the timestamp pallet.
+#[derive_impl(pallet_timestamp::config_preludes::TestDefaultConfig)]
```

Comment:
We should not implement the test defaults here and for all the other pallets. People will just blindly copy this.

Reply:
> We should not implement the test defaults here and for all the other pallets. People will just blindly copy this.

Got it. @kianenigma We should update the minimal template as well then?

Reply:
I've shared my opinion on this before, so just repeating it here: 

1. For tests, we should override everything.
2. For real runtimes, I would personally still provide and use defaults. There was push back from others, but I think this all flies against our important goal of improving developer experience, but I can agree to disagree here.  
3. Even if you want to be more conservative and not do what I, what we discussed elsewhere was to find a way to express defaults just for trivial things. I see zero harm in providing defaults for e.g. `type RuntimeEvent = RuntimeEvent`, and other configs that are equally trivial.

Reply:
https://github.com/paritytech/polkadot-sdk/pull/4056#discussion_r1644081212

Reply:
Removed these changes for now. Will take this up in a separate PR.